bicep/main.bicep

```diff
@@ -171,21 +170,17 @@ var appGwSubnetId = ingressApplicationGateway ? (custom_vnet ? network.outputs.a
 
 @description('The full Azure resource ID of the DNS zone to use for the AKS cluster')
 param dnsZoneId string = ''
-var dnsZoneRg = !empty(dnsZoneId) ? split(dnsZoneId, '/')[4] : ''
-var dnsZoneName = !empty(dnsZoneId) ? split(dnsZoneId, '/')[8] : ''
-var isPrivate = !empty(dnsZoneId) ? split(dnsZoneId, '/')[7] == 'privateDnsZones' : false
+var isDnsZonePrivate = !empty(dnsZoneId) ? split(dnsZoneId, '/')[7] == 'privateDnsZones' : false
 
-module dnsZone './dnsZone.bicep' = if (!empty(dnsZoneId)) {
+module dnsZone './dnsZoneRbac.bicep' = if (!empty(dnsZoneId)) {
   name: 'addDnsContributor'
```

Comment:
you have this deployment name in 2 places, better to make them distinct so can track it in the portal deployment panel